### PR TITLE
Use existing sm in tests

### PR DESCRIPTION
**Pull request by rajatagarwal-ibm**
### Description

Use permanent SM instance account for tests

### Release required?

- [x] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: tests/scripts/pre-validation.sh [hunk 1]
```diff
@@ -10,6 +10,7 @@ DA_DIR="solutions/banking"
 TERRAFORM_SOURCE_DIR="tests/resources/existing-resources"
 JSON_FILE="${DA_DIR}/catalogValidationValues.json"
 REGION="us-south"
+SECRETS_MANAGER_GUID="79c6d411-c18f-4670-b009-b0044a238667"
```
Comment (ocofaigh) [inline tests/scripts/pre-validation.sh:13]: can you pull this from the permanent resource yaml instead of hard coding it incase it ever changes

**Review by ocofaigh — CHANGES_REQUESTED** [review]
1 comment

File: tests/scripts/pre-validation.sh [hunk 2]
```diff
@@ -10,6 +10,7 @@ DA_DIR="solutions/banking"
 TERRAFORM_SOURCE_DIR="tests/resources/existing-resources"
 JSON_FILE="${DA_DIR}/catalogValidationValues.json"
 REGION="us-south"
+SECRETS_MANAGER_GUID=$(curl -X GET --retry 3 -fLsS https://raw.githubusercontent.com/terraform-ibm-modules/common-dev-assets/main/common-go-assets/common-permanent-resources.yaml | yq e '.secretsManagerGuid')
```
Comment (ocofaigh) [inline tests/scripts/pre-validation.sh:13]: no need to do this either - common-dev-assets will be locally available in the git submodule - so just pull it locally

File: tests/scripts/pre-validation.sh [hunk 3]
```diff
@@ -24,7 +26,7 @@ TF_VARS_FILE="terraform.tfvars"
     echo "region=\"${REGION}\""
     echo "prefix=\"${PREFIX}\""
   } >> ${TF_VARS_FILE}
-  terraform apply -input=false -auto-approve -var-file=${TF_VARS_FILE} || exit 1
+    terraform apply -input=false -auto-approve -var-file=${TF_VARS_FILE} || exit 1
```
Comment (ocofaigh) [inline tests/scripts/pre-validation.sh:29]: no need to indent here

**Review by ocofaigh — APPROVED** [review]

**Review by vburckhardt — APPROVED** [review]

**Pull request merged**